test/test_expression.cpp

```diff
@@ -76,7 +76,8 @@ TEST_CASE("Expression")
         REQUIRE(exp_t::get_precedence(FUN_CALL) >= exp_t::get_precedence(ARRAY));
         REQUIRE(exp_t::get_precedence(ARRAY) >= exp_t::get_precedence(DOT));
         REQUIRE(exp_t::get_precedence(DOT) >= exp_t::get_precedence(RATE));
-        REQUIRE(exp_t::get_precedence(RATE) > exp_t::get_precedence(PRE_INCREMENT));
+        REQUIRE(exp_t::get_precedence(PRE_INCREMENT) == exp_t::get_precedence(NOT));
+        REQUIRE(exp_t::get_precedence(NOT) < exp_t::get_precedence(ARRAY));
```

Comment:
Please keep the order and use greater-than operator (it makes it easier to check against https://en.cppreference.com/w/c/language/operator_precedence).
In particular this:
```c++
REQUIRE(exp_t::get_precedence(NOT) < exp_t::get_precedence(ARRAY));
```
Should be:
```c++
REQUIRE(exp_t::get_precedence(ARRAY) > exp_t::get_precedence(NOT));
```
and moved up to where `REQUIRE(exp_t::get_precedence(ARRAY)...`  is.
As you can see, this check becomes redundant as `ARRAY` is high in this list and `NOT` is below.

Otherwise we are doomed to keep enumerating all pairs.

Reply:
I've made the assertion more consistent with the others